python/dtype.cpp

```diff
@@ -84,6 +82,10 @@ parse_datetime_dtype(const std::string_view dtype_name) {
       throw std::invalid_argument(
           std::string("Unsupported unit in datetime: ") + dtype_name[11] + "s");
     }
+  } else if (length == 10) {
+    if (dtype_name == "datetime64") {
+      return scipp::units::dimensionless;
+    }
```

Comment:
Would this function be better written and easier to understand/debug if we used `std::regex` for matches rather than lengths and character comparisons?

Reply:
Could the logic here be moved/extracted in such a way we could put the c++ parsing under test?  I can see you have added tests on the python side - maybe this is sufficient.

Reply:
The main reason for avoiding `std::regex` is that it cannot handle single chars but always gives you c-strings. This seems overkill when you try  to identify individual chars like 's'. But it is probably overkill, this code should not be performance relevant.

Concerning testing, this would mean moving it out of the python bindings library. Not sure this makes a lot of sense.

Reply:
👍  on not testing binding code directly.